platform/darwin/src/MGLFeature.h

```diff
@@ -148,7 +148,7 @@ NS_ASSUME_NONNULL_BEGIN
  when the feature instance is used to initialize an `MGLShapeSource` and that
  source is added to the map and styled.
  */
-@property (nonatomic, copy) NSDictionary<NSString *, id> *attributes;
+@property (nonatomic, copy, nullable) NSDictionary<NSString *, id> *attributes;
```

Comment:
This changes how `attributes` bridges in Swift, so code that previously would’ve said `attributes["name"]` would have to be changed to `attributes?["name"] ?? "…"`.

Would it be possible instead to set this property to an empty dictionary on initialization as a fallback? That would also avoid the crash.

Reply:
@1ec5 how expensive would it be to create empty dictionary for every feature?

Reply:
Good point, creating an empty dictionary for each feature could result in quite a bit of waste for some data sets. Since this property has copy semantics, rather than retain semantics, how about customizing the getter to create and return an empty dictionary on demand as a fallback when the underlying dictionary would be nil?

Reply:
I like the idea of a custom getter. This is a protocol though right? So we'd be relying on implementers to do that (which may not be an issue at the moment)?

Reply:
Yes, it’ll require a separate implementation in each of the MGLFeature classes until we fix #7454. A macro that implements the entire getter (including the method signature) would come in handy here.

Reply:
In the immediate short-term, how do you feel about removing the `nullable` specifier, so that the public API doesn't change? (Yet we keep the nil check)